astropy/utils/data.py

```diff
@@ -244,16 +246,49 @@ def get_readable_fileobj(name_or_obj, encoding=None, cache=False,
             delete_fds.append(tmp)
             fileobj_new = bz2.BZ2File(tmp.name, mode='rb')
             fileobj_new.read(1)  # need to check that the file is really bzip2
-        except IOError:  # invalid bzip2 file
+        except (IOError) as e:  # invalid bzip2 file
```

Comment:
Right, blindly copied that over from the xz case. And currently the `e` is not needed, as I have for now commented the line propagating it either as `IOError` or `ValueError` - there seem to be too many use cases of `ascii.read` relying on no error being raised on invalid files...